snuba/datasets/plans/storage_builder.py

```diff
@@ -92,112 +98,79 @@ def get_query_data_source(
-        ):
-            # The QueryTranslator class should be instantiated once for each call to build_plan,
-            # to avoid cache conflicts.
-            clickhouse_query = QueryTranslator(self.__mappers).translate(query)
-
-        with sentry_sdk.start_span(
-            op="build_plan.single_storage", description="set_from_clause"
-        ):
-            clickhouse_query.set_from_clause(
-                get_query_data_source(
-                    self.__storage.get_schema().get_data_source(),
-                    final=query.get_final(),
-                    sampling_rate=query.get_sample(),
-                )
-            )
-
-        cluster = self.__storage.get_cluster()
-
-        db_query_processors = [
-            *self.__storage.get_query_processors(),
-            *self.__post_processors,
-            MandatoryConditionApplier(),
-            MandatoryConditionEnforcer(
-                self.__storage.get_mandatory_condition_checkers()
-            ),
-        ]
-
-        return [
-            ClickhouseQueryPlan(
-                query=clickhouse_query,
-                plan_query_processors=[],
-                db_query_processors=db_query_processors,
-                storage_set_key=self.__storage.get_storage_set_key(),
-                execution_strategy=SimpleQueryPlanExecutionStrategy(
-                    cluster=cluster,
-                    db_query_processors=db_query_processors,
-                    splitters=self.__storage.get_query_splitters(),
-                ),
-            )
-        ]
-
-
-class SelectedStorageQueryPlanBuilder(ClickhouseQueryPlanBuilder):
+class StorageQueryPlanBuilder(ClickhouseQueryPlanBuilder):
     """
-    A query plan builder that selects one of multiple storages in the dataset.
+    Builds the Clickhouse Query Execution Plan for a dataset which supports single storages,
+    multiple storages, unsliced storages, and sliced storages.
     """
 
     def __init__(
         self,
-        selector: QueryStorageSelector,
+        storages: List[StorageAndMappers],
+        selector: Optional[QueryStorageSelector] = None,
         post_processors: Optional[Sequence[ClickhouseQueryProcessor]] = None,
+        partition_key_column_name: Optional[str] = None,
     ) -> None:
+        self.__storages = storages
         self.__selector = selector
         self.__post_processors = post_processors or []
+        self.__partition_key_column_name = partition_key_column_name
 
     @with_span()
     def build_and_rank_plans(
         self, query: LogicalQuery, settings: QuerySettings
     ) -> Sequence[ClickhouseQueryPlan]:
-        with sentry_sdk.start_span(
-            op="build_plan.selected_storage", description="select_storage"
-        ):
-            storage, mappers = self.__selector.select_storage(query, settings)
+        if len(self.__storages) < 1:
+            raise QueryStorageSelectorError("No storages specified to select from.")
+
+        if not self.__selector:
```

Comment:
This should only be used by entities that are defined in YAML correct? So this check would be removed once we support storage selection in config?

Reply:
Right, although jsonschema syntax validation should complain before this gets raised.